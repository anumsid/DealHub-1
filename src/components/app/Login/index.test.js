import React from 'react'
import { shallow } from 'enzyme'
import { stub } from 'sinon'
import toJson from 'enzyme-to-json'
import { SignIn } from './index'

describe('Login Component', () => {
  let wrapper
  let onClick
  beforeEach(() => {
    onClick = stub()
    const props = {
      errorMessage: null,
      onClick: onClick
    }
    wrapper = shallow(<SignIn {...props} />)
  })

  it('Matches snapshot', () => {
    expect(toJson(wrapper)).toMatchSnapshot()
  })

  it('Handles click on form submit', () => {
    wrapper.find('FormInput[type="email"]').simulate('change', {target: {value: 'test@test.com', getAttribute: () => 'email'}})
    wrapper.find('FormInput[type="password"]').simulate('change', {target: {value: 'password', getAttribute: () => 'password'}})
    wrapper.find('FormButton').simulate('click')
    expect(onClick.calledOnce).toBe(true)
  })
})
