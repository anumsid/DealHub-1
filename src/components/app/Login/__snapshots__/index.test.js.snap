// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login Component Matches snapshot 1`] = `
<div
  className="signInPage"
>
  <div>
    <h3
      id="slogan"
    >
      Real deals crowdsourced by an awesome community 
    </h3>
  </div>
  <br />
  <br />
  <br />
  <Form
    as="form"
    className="mx-auto"
    style={
      Object {
        "width": "40rem",
      }
    }
  >
    <FormInput
      as={[Function]}
      control={[Function]}
      fluid={true}
      label="Enter email"
      onChange={[Function]}
      placeholder="Email"
      type="email"
    />
    <FormInput
      as={[Function]}
      control={[Function]}
      fluid={true}
      label="Enter password"
      onChange={[Function]}
      placeholder="Password"
      type="password"
    />
    <br />
    <br />
    <FormButton
      as={[Function]}
      control={[Function]}
      onClick={[Function]}
    >
      Login
    </FormButton>
  </Form>
</div>
`;
